src/main/scala/viper/gobra/translator/encodings/typeless/AssertionEncoding.scala

```diff
@@ -54,6 +54,13 @@ class AssertionEncoding extends Encoding {
         case Seq() => newExists
         case errors => Violation.violation(s"invalid trigger pattern (${errors.head.readableMessage})")
       }
+
+    case let: in.Let =>
+      for {
+        exp <- ctx.expression(let.in)
+        l = ctx.variable(let.left)
+        r <- ctx.expression(let.right)
+      } yield withSrc(vpr.Let(l, r, exp), let)
```

Comment:
The writer also has features to introduce let bindings, for instance `_ <- bind(l, exp)`, but this encoding should also be fine.